engine.py

```diff
@@ -1041,18 +1041,48 @@ def _win32_get_proxy_window(self):
         # Get the main Photoshop window:
         ps_hwnd = self._win32_get_photoshop_main_hwnd()
         win32_proxy_win = None
+        proxy_win_hwnd = None
 
         if ps_hwnd:
             from tank.platform.qt import QtGui, QtCore
 
             # Create the proxy QWidget.
             win32_proxy_win = QtGui.QWidget()
-            win32_proxy_win.setWindowTitle('sgtk dialog owner proxy')
+            window_title = "sgtk dialog owner proxy"
+            win32_proxy_win.setWindowTitle(window_title)
+
+            # We have to take different approaches depending on whether
+            # we're using Qt4 (PySide) or Qt5 (PySide2). The functionality
+            # needed to turn a Qt5 WId into an HWND is not exposed in PySide2,
+            # so we can't do what we did below for Qt4.
+            if QtCore.__version__.startswith("4."):
+                proxy_win_hwnd = self.__tk_photoshopcc.win_32_api.qwidget_winid_to_hwnd(
+                    win32_proxy_win.winId(),
+                )
+            else:
+                # With PySide2, we're required to look up our proxy parent
+                # widget's HWND the hard way, following the same logic used
+                # to find Photoshop's main window. To do that, we actually have
+                # to show our widget so that Windows knows about it. We can make
+                # it effectively invisible if we zero out its size, so we do that,
+                # show the widget, and then look up its HWND by window title before
+                # hiding it.
+                win32_proxy_win.setGeometry(0, 0, 0, 0)
+                win32_proxy_win.show()
 
-            proxy_win_hwnd = self.__tk_photoshopcc.win_32_api.qwidget_winid_to_hwnd(
-                win32_proxy_win.winId(),
-            )
+                try:
+                    proxy_win_hwnd_found = self.__tk_photoshopcc.win_32_api.find_windows(
+                        stop_if_found=True,
+                        window_text=window_title,
+                    )
+                finally:
+                    win32_proxy_win.hide()
+
+                if proxy_win_hwnd_found:
+                    proxy_win_hwnd = proxy_win_hwnd_found[0]
 
+        # Parent to the Photoshop application window.
+        if proxy_win_hwnd is not None:
```

Comment:
What happens if `proxy_win_hwnd` is `None`? Is it better to raise a RuntimeError if `proxy_win_hwnd` is None and skip this if clause altogether?

Reply:
Well, not figuring out the Window parenting stuff here is recoverable in a way that Toolkit apps will still function properly. However, I'm adding a warning message so it's clear in the logs why window parenting isn't working if we ever hit a support situation where it comes up.